src/main/java/com/google/neighborgood/servlets/EditTaskServlet.java

```diff
@@ -80,15 +87,35 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         System.err.println("Unable to find the entity based on the input key");
         response.sendError(
             HttpServletResponse.SC_NOT_FOUND, "The requested task could not be found");
+        return;
 
       } finally {
         if (transaction.isActive()) {
           transaction.rollback();
           System.err.println("Task must be open to be claimed by a helper");
           response.sendError(
               HttpServletResponse.SC_CONFLICT, "Task has already been claimed by another helper");
+          return;
         }
       }
+      // When a user takes a new task, send the user notifications for the messages that the owner
```

Comment:
Maybe `doPost` is due for a bit of refactoring. I think it would be easier to read if it looked something like:
```java
    String keyString = request.getParameter("task-id");
    Key taskKey = KeyFactory.stringToKey(keyString);
    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
    Entity task;

    UserService userService = UserServiceFactory.getUserService();
    if (!userService.isUserLoggedIn()) {
      System.err.println("User must be logged in to edit a task");
      response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
      response.sendError(
          HttpServletResponse.SC_UNAUTHORIZED,
          "You must be logged in to perform this action on a task");
      return;
    }

    claimTasksForUser();

    getPastNotificationsForClaimedTasks();
```

Reply:
Sure. I would modify the part of code that gets past notifications for the helper of the newly-claimed task, and separate it into a function `getPastNotificationsForClaimedTasks()` in the new commit. And @poncem91 wrote the part of code that claims tasks for user. So maybe she could help me with separating the part of code that claims tasks for user into a separate function.

Reply:
We should definitely tackle this when we go over our whole codebase as part of our 'consolidate' issue. We should probably start with that tomorrow or Thursday?

Reply:
SGTM :)